Build x86_64-pc-windows-gnu target on Linux fails

```
[lion328@lion328-LINUXPC sys-info-rs]$ cargo build --target=x86_64-pc-windows-gnu
   Compiling libc v0.2.32
   Compiling gcc v0.3.54
   Compiling sys-info v0.5.1 (file:///mnt/40daab41-cd89-4a57-9ae2-95ae141940d9/GitRepos/sys-info-rs)
warning: use of deprecated item
 --> build.rs:5:9
  |
5 |         gcc::compile_library("libinfo.a", &["c/linux.c"]);
  |         ^^^^^^^^^^^^^^^^^^^^
  |
  = note: #[warn(deprecated)] on by default

warning: use of deprecated item
 --> build.rs:7:9
  |
7 |         gcc::compile_library("libinfo.a", &["c/macos.c"]);
  |         ^^^^^^^^^^^^^^^^^^^^

warning: use of deprecated item
 --> build.rs:9:9
  |
9 |         gcc::compile_library("libinfo.a", &["c/windows.c"]);
  |         ^^^^^^^^^^^^^^^^^^^^

error: failed to run custom build command for `sys-info v0.5.1 (file:///mnt/40daab41-cd89-4a57-9ae2-95ae141940d9/GitRepos/sys-info-rs)`
process didn't exit successfully: `/mnt/40daab41-cd89-4a57-9ae2-95ae141940d9/GitRepos/sys-info-rs/target/debug/build/sys-info-3295ef534ba2f7db/build-script-build` (exit code: 101)
--- stdout
TARGET = Some("x86_64-pc-windows-gnu")
OPT_LEVEL = Some("0")
TARGET = Some("x86_64-pc-windows-gnu")
HOST = Some("x86_64-unknown-linux-gnu")
TARGET = Some("x86_64-pc-windows-gnu")
TARGET = Some("x86_64-pc-windows-gnu")
HOST = Some("x86_64-unknown-linux-gnu")
CC_x86_64-pc-windows-gnu = None
CC_x86_64_pc_windows_gnu = None
TARGET_CC = None
CC = None
HOST = Some("x86_64-unknown-linux-gnu")
CROSS_COMPILE = None
TARGET = Some("x86_64-pc-windows-gnu")
HOST = Some("x86_64-unknown-linux-gnu")
CFLAGS_x86_64-pc-windows-gnu = None
CFLAGS_x86_64_pc_windows_gnu = None
TARGET_CFLAGS = None
CFLAGS = None
DEBUG = Some("true")
running: "x86_64-w64-mingw32-gcc" "-O0" "-ffunction-sections" "-fdata-sections" "-g" "-m64" "-Wall" "-Wextra" "-o" "/mnt/40daab41-cd89-4a57-9ae2-95ae141940d9/GitRepos/sys-info-rs/target/x86_64-pc-windows-gnu/debug/build/sys-info-d0550d93732dd07b/out/c/linux.o" "-c" "c/linux.c"
cargo:warning=c/linux.c:4:10: fatal error: sys/sysinfo.h: No such file or directory
cargo:warning= #include <sys/sysinfo.h>
cargo:warning=          ^~~~~~~~~~~~~~~
cargo:warning=compilation terminated.
exit code: 1

--- stderr
thread 'main' panicked at '

Internal error occurred: Command "x86_64-w64-mingw32-gcc" "-O0" "-ffunction-sections" "-fdata-sections" "-g" "-m64" "-Wall" "-Wextra" "-o" "/mnt/40daab41-cd89-4a57-9ae2-95ae141940d9/GitRepos/sys-info-rs/target/x86_64-pc-windows-gnu/debug/build/sys-info-d0550d93732dd07b/out/c/linux.o" "-c" "c/linux.c" with args "x86_64-w64-mingw32-gcc" did not execute successfully (status code exit code: 1).

', /home/lion328/.cargo/registry/src/github.com-1ecc6299db9ec823/gcc-0.3.54/src/lib.rs:1670:4
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

As you can see its build `c/linux.c` instead of `c/windows.c`. This might be caused by `build.rs` from using `cfg!(target_os = "...")` to tell the target OS.

--- a/build.rs
+++ b/build.rs
@@ -1,14 +1,18 @@
 extern crate gcc;
 
+use std::env;
+
 fn main() {
-    if cfg!(target_os = "linux") {
-        gcc::compile_library("libinfo.a", &["c/linux.c"]);
-    } else if cfg!(target_os = "macos") {
-        gcc::compile_library("libinfo.a", &["c/macos.c"]);
-    } else if cfg!(target_os = "windows") {
-        gcc::compile_library("libinfo.a", &["c/windows.c"]);
-        println!("cargo:rustc-flags=-l psapi");
-    } else {
-        panic!("Unsupported system");
-    }
+    let target = env::var("TARGET").unwrap();
+    let target_os = target.split('-').nth(2).unwrap();
+
+    match target_os {
+        "linux" => gcc::compile_library("libinfo.a", &["c/linux.c"]),
+        "darwin" => gcc::compile_library("libinfo.a", &["c/macos.c"]),
+        "windows" => {
+            gcc::compile_library("libinfo.a", &["c/windows.c"]);
+            println!("cargo:rustc-flags=-l psapi");
+        },
+        _ => panic!("Unsupported system")
+    };
 }

